File: easy/multiply_lists.rb
File.open(ARGV[0]).each_line do |line|
  s = line.delete('|').split.map(&:to_i)
  0.upto(s.length / 2 - 1) do |i|
    print ' ' if i > 0
    print s[i] * s[s.length / 2 + i]
  end
  puts
end

File: easy/multiply_lists.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $s, $t ) = split( / \| /, $line );
    my @s = split( / /, $s );
    my @t = split( / /, $t );
    for ( my $i = 0 ; $i < scalar @s ; $i++ ) {
        @s[$i] *= @t[$i];
    }
    printf( "%s\n", join( ' ', @s ) );
}
close(INFILE);
